build-gcc-7.3.0.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Our directory and file names.
+GCC_VERSION='7.3.0'
+GCC_TMP_DIR='gcc-tmp'
+GCC_BUILD_DIR='gcc-build'
+GCC_INSTALL_DIR='gcc-install'
+GCC_PKG_TARBALL='gcc.tar.gz'
+
+# Configuration for this repository.
+GCC_SUPPORTDIR='build-support/bin/gcc'
+
+# Shared configuration across supported platforms.
+GCC_SUPPORTED_LANGS='c,c++,objc,obj-c++,fortran'
+
+# Name of the resulting directory when we extract the tarball.
+GCC_SRC_DIRNAME="gcc-${GCC_VERSION}"
+GCC_RELEASE_ARCHIVE_FILE="${GCC_SRC_DIRNAME}.tar.gz"
+GCC_RELEASE_URL="https://ftpmirror.gnu.org/gnu/gcc/gcc-${GCC_VERSION}/${GCC_RELEASE_ARCHIVE_FILE}"
+
+CONFIGURE_BASE_ARGS=(
+  --disable-multilib
+  --enable-languages="${GCC_SUPPORTED_LANGS}"
+  --enable-checking='release'
+  --with-pkgversion="Pants-packaged GCC (${GCC_VERSION})"
+  --with-bugurl='https://github.com/pantsbuild/pants/issues'
+)
+
+LINUX_SUPPORTED_ARCHS=(
+  x86_64
+)
+
+OSX_SUPPORTED_VERSIONS=(
+  10.7
+  10.8
+  10.9
+  10.10
+  10.11
+  10.12
+  10.13
+)
+
+# Default to -j2
+MAKE_JOBS="${MAKE_JOBS:-2}"
+
+## Set any computed and/or platform-specific variables.
+
+configure_args_for_current_platform=("${CONFIGURE_BASE_ARGS[@]}")
+
+build_support_directories_for_current_platform=()
+
+_host_uname="$(uname)"
+
+function die {
+  echo >&2 "$@"
+  exit 1
+}
+
+case "$_host_uname" in
+  Darwin)
+    # Parallel builds fail on High Sierra. Homebrew has a patch for it:
+    # https://github.com/Homebrew/homebrew-core/blob/622c201de21eb84677143822044dc8955c01dc3a/Formula/gcc.rb#L56
+    # but I haven't reviewed that patch or the corresponding gcc bug:
+    # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=81797
+    # enough to blindly use the Homebrew patch.
+    MAKE_JOBS=1
+
+    unset LD
+    unset LDSHARED
+    PATH='/usr/bin:/bin'
+
+    _osx_uname_release="$(uname -r)"
+    configure_args_for_current_platform+=(
+      --build="x86_64-apple-darwin${_osx_uname_release}"
+      --host="x86_64-apple-darwin${_osx_uname_release}"
+      --target="x86_64-apple-darwin${_osx_uname_release}"
+      --with-system-zlib
```

Comment:
Is statically linking zlib an option? Fewer system deps are better imo...

Reply:
Agreed, that was ripped from the homebrew recipe without further thought. I have since tried investigating how difficult it would be to build all of these from the CentOS docker image, which changed these again (this file is now  `build-gcc.sh`).